src/chunkserver/hddspacemgr.cc

```diff
@@ -2510,97 +2510,6 @@ void hddTerminate(void) {
 	gDisks.clear();
 }
 
-int hddSizeParse(const char *str, uint64_t *ret) {
-	TRACETHIS();
-	uint64_t val, frac, fracdiv;
-	double drval, mult;
-	int f;
-	val = 0;
-	frac = 0;
-	fracdiv = 1;
-	f = 0;
-	while (*str >= '0' && *str <= '9') {
-		f = 1;
-		val *= 10;
-		val += (*str - '0');
-		str++;
-	}
-	if (*str == '.') { // accept format ".####" (without 0)
-		str++;
-		while (*str >= '0' && *str <= '9') {
-			fracdiv *= 10;
```

Comment:
Maybe low priority issue, but ```frac``` and ```fracdiv``` may overflow when the number of digits after the dot increases to around 20 (the value of ```fracdiv``` should be 10^20 which overflows in uint64_t).

Probably you could use ```long double``` for large precission or other floating point types to get through it.